docker-compose.yml

```diff
@@ -0,0 +1,23 @@
+version: "3.9"
+   
+services:
+  db:
+    image: mysql:8
+    command: mysqld --default-authentication-plugin=mysql_native_password
+    restart: always
+    volumes:
+      - imdb_data:/var/lib/mysql
+    env_file: 
+      - ./db.secret.env
+  web:
+    build: .
+    entrypoint: ['./wait-for-it.sh', 'db:3306', '--', './web_entrypoint.sh'] 
```

Comment:
This is rather complex for a simple app. Minimize this complex app.

Reply:
I've simplified the startup by removing the shell scripts. Should I remove the `db.secret.env` file too? It currently has the following contents:
```bash
MYSQL_ROOT_PASSWORD=secret
MYSQL_DATABASE=imdb
```

Reply:
Why was this file needed? You can use environment variables in docker-compose to pass any environment-related information.

Reply:
The repository is public so I thought I should follow the recommended practices. I read in many places that checking-out environment variables can be risky because we normally use env vars for secrets. For example, [this article](https://medium.com/better-programming/using-variables-in-docker-compose-265a604c2006) says:

> Storing the values of your environment variables in the Compose file, which, nine and a half times out of ten, will go to straight to source control, is a huge security risk. Luckily, we have an alternative: using an external file to store our environment variables.

It is a practice project so maybe it is unnecessary. I just wanted to try out the option for learning. :p

Reply:
First thing, the secrets are never public. The environment variable(as the name suggests) can vary based on the environment. The production env will have different values as dev. Having an env var or setting var is to keep the code consistent but have the values vary in different environments.

Reply:
Oh, I understand now. You mean like this, right?

```yaml
services:
  db:
    environment: 
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
```

I'll update the code, this approach seems better. Thanks.